sources/common/toddl.go

```diff
@@ -71,10 +82,125 @@ func (ss *SchemaToSpannerImpl) SchemaToSpannerDDL(conv *internal.Conv, toddl ToD
+
+			expressionDetail := internal.ExpressionDetail{
+				Expression:       cc.Expr,
+				Type:             "CHECK",
+				ReferenceElement: internal.ReferenceElement{Name: sp.Name},
+				ExpressionId:     cc.ExprId,
+				Metadata:         map[string]string{"tableId": sp.Id},
+			}
+			expressionDetailList = append(expressionDetailList, expressionDetail)
+		}
+	}
+
+	return expressionDetailList
+}
+
+// removeCheckConstraint this method will remove the constraint which has error
+func RemoveCheckConstraints(checkConstraints []ddl.CheckConstraint, exprVerificationOutputs []internal.ExpressionVerificationOutput) []ddl.CheckConstraint {
+	unwantedExprIds := make(map[string]bool)
+	for _, ev := range exprVerificationOutputs {
+		if !ev.Result {
+			unwantedExprIds[ev.ExpressionDetail.ExpressionId] = true
+		}
+	}
+
+	outIdx := 0
+	for _, checkConstraint := range checkConstraints {
+		if !unwantedExprIds[checkConstraint.ExprId] {
+			checkConstraints[outIdx] = checkConstraint
+			outIdx++
+		}
+	}
+	return checkConstraints[:outIdx]
+}
+
+// VerifyExpression this function will use expression_api to validate check constraint expressions and add the relevant error
+// to suggestion tab and remove the check constraint which has error
+func (ss *SchemaToSpannerImpl) VerifyExpressions(conv *internal.Conv) error {
+	ctx := context.Background()
+
+	spschema := conv.SpSchema
+
+	verifyExpressionsInput := internal.VerifyExpressionsInput{
+		Conv:                 conv,
+		Source:               conv.Source,
+		ExpressionDetailList: GenerateExpressionDetailList(spschema),
+	}
+
+	result := ss.ExpressionVerificationAccessor.VerifyExpressions(ctx, verifyExpressionsInput)
+	if result.ExpressionVerificationOutputList == nil {
+		return result.Err
+	}
+
+	exprOutputsByTable := make(map[string][]internal.ExpressionVerificationOutput)
+	for _, ev := range result.ExpressionVerificationOutputList {
+		if !ev.Result {
+			tableId := ev.ExpressionDetail.Metadata["tableId"]
+			exprOutputsByTable[tableId] = append(exprOutputsByTable[tableId], ev)
+		}
+	}
+
+	for tableId, exprOutputs := range exprOutputsByTable {
+
+		spschema := conv.SpSchema[tableId]
+		spschema.CheckConstraints = RemoveCheckConstraints(spschema.CheckConstraints, exprOutputs)
+		conv.SpSchema[tableId] = spschema
+
+		for _, ev := range exprOutputs {
+
+			var issueType internal.SchemaIssue
+
+			for key, issue := range ErrorTypeMapping {
+				if strings.Contains(ev.Err.Error(), key) {
+					issueType = issue
+					break
+				}
+			}
+
+			if _, exists := conv.SchemaIssues[tableId]; !exists {
+				conv.SchemaIssues[tableId] = internal.TableIssues{
+					TableLevelIssues: []internal.SchemaIssue{},
+				}
+			}
+
+			tableIssue := conv.SchemaIssues[tableId]
+
+			if !IsSchemaIssuePresent(tableIssue.TableLevelIssues, issueType) {
+				tableIssue.TableLevelIssues = append(tableIssue.TableLevelIssues, issueType)
+			}
+			conv.SchemaIssues[tableId] = tableIssue
+		}
+	}
+
+	return nil
```

Comment:
This function always returns a nil error

Reply:
this function will call the expression API; it will throw an unhandled error and modify the conv object.